### PR TITLE
NSB.Metrics for NServiceBus 7.0

### PR DESCRIPTION
**Pull request by Scooletz**
This PR moves `NServiceBus.Metrics` to NSB v7.0. Beside migration, it removes custom reporting capabilities and looses dependency on Metrics.NET. With version 3.0 we've got a library providing a set of probes that one can register to. 

Additionally, it introduces a test suite that covers all the probes, even future ones. The suite adds two tests, for a successful scenario and a scenario where a handler throws an exception. For the second one, 1 immediate retry is configured.

### CODE REVIEW

**Comment by Scooletz** [comment]
Ping @Particular/metrics-maintainers This is ready for a review. Once we have it we can proceed with marking it as release and move forward with releasing this cohort of packages ;-)

File: src/NServiceBus.Metrics/MetricsFeature.cs [hunk 1]
```diff
@@ -15,50 +14,15 @@ protected override void Setup(FeatureConfigurationContext context)
...
         var settings = context.Settings;
         var options = settings.Get<MetricsOptions>();
-        var endpointName = settings.EndpointName();
 
         SetUpRegisteredObservers(context, options, probeContext);
-
-        SetUpLegacyReporters(context, options, endpointName, probeContext);
     }
 
     static void SetUpRegisteredObservers(FeatureConfigurationContext context, MetricsOptions options, ProbeContext probeContext)
```
Comment (tmasternak) [inline src/NServiceBus.Metrics/MetricsFeature.cs]: Can we inline this method. I'm not sure it brings value now when it's the only one called in feature `Setup`

File: src/NServiceBus.Metrics/NServiceBus.Metrics.csproj [hunk 2]
```diff
@@ -15,23 +15,8 @@
   </PropertyGroup>
   <ItemGroup>
     <PackageReference Include="ILRepack" Version="2.*" PrivateAssets="All" />
```
Comment (tmasternak) [inline src/NServiceBus.Metrics/NServiceBus.Metrics.csproj]: Why do we still need `ILRepack`?

**Comment by Scooletz** [comment]
Ping @tmasternak

**Comment by seanfarmar** [comment]
@Scooletz @tmasternak can we merge this?

**Comment by Scooletz** [comment]
@seanfarmar As soon as the original reviewer @tmasternak agrees ;-)

**Comment by SimonCropp** [comment]
so looking at the current code in this branch. i can see no tests that verify anything. am i missing something?

**Comment by tmasternak** [comment]
@SimonCropp what kind of verification do you mean?

**Comment by SimonCropp** [comment]
@tmasternak unit tests to verify the code executes as expected

**Comment by SimonCropp** [comment]
eg not tests are run here https://builds.particular.net/viewLog.html?buildId=238369&buildTypeId=NServiceBus_NServiceBusMetrics_3TestNetFrameworkOnWindows&tab=testsInfo

**Comment by Scooletz** [comment]
I'm WIPping it again, till I got these tests right.

**Comment by Scooletz** [comment]
Ping @Particular/metrics-maintainers I need some help with this strong naming failing on acceptance tests. Once, it's done, it's ready to review.

**Comment by Scooletz** [comment]
Cool. So we've got a green build. Ping @Particular/metrics-maintainers for a review.

File: src/NServiceBus.Metrics.AcceptanceTests/Tests/When_having_metrics_handlers_registered.cs [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+using System.Threading.Tasks;
+using NServiceBus;
+using NServiceBus.AcceptanceTesting;
+using NServiceBus.AcceptanceTests;
+using NServiceBus.AcceptanceTests.EndpointTemplates;
+using NUnit.Framework;
+
+public class When_having_metrics_handlers_registered : NServiceBusAcceptanceTest
+{
+    readonly HashSet<string> positiveProbes = new HashSet<string>
```
Comment (SimonCropp) [inline src/NServiceBus.Metrics.AcceptanceTests/Tests/When_having_metrics_handlers_registered.cs]: i would remove `readonly` as it is only noise in this case

**Pull request merged**